pkg/util/stop/stopper.go

```diff
@@ -543,24 +523,25 @@ func (s *Stopper) IsStopped() <-chan struct{} {
 // tasks complete. This is used from Stop() and unittests.
 func (s *Stopper) Quiesce(ctx context.Context) {
 	defer s.Recover(ctx)
-	s.mu.Lock()
-	defer s.mu.Unlock()
+	if func() bool {
+		s.mu.Lock()
+		defer s.mu.Unlock()
+		if !s.mu.quiescing {
+			s.mu.quiescing = true
+			close(s.quiescer)
+			return true
+		}
+		return false
+	}() {
+		log.Infof(ctx, "quiescing")
```

Comment:
Instead of this `if func() {...}` structure, if all we're doing is logging when quiescing for the first time, something like the following would be clearer:
```
func() {
  // ...
  // log when needed
}()
```